hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/ITestS3AFileOperationCost.java

```diff
@@ -258,6 +259,33 @@ public void testCostOfGetFileStatusOnMissingFile() throws Throwable {
         GET_FILE_STATUS_FNFE);
   }
 
+  @Test
+  public void testCostOfRootFileStatus() throws Throwable {
+    Path root = path("/");
+    S3AFileStatus rootStatus = verifyRawInnerGetFileStatus(
+            root,
+            false,
+            StatusProbeEnum.ALL,
+            ROOT_FILE_STATUS_PROBE);
+    Assertions.assertThat(rootStatus.isDirectory())
```

Comment:
for strictness, how about before L270 you do a status.toString() into a variable, and include that at the end of each describedAs message. That way, if there really was an assertion failure, we'd see the status